File: nom-ui/src/app/recipe/models/recipe-edit.model.ts
// File: nom-ui/src/app/recipe/models/recipe-edit.model.ts

import { RecipeIngredientModel } from './recipe-ingredient.model';
import { RecipeStepModel } from './recipe-step.model';

export interface RecipeEditModel {
    id: number;
    name: string;
    description?: string;
    ingredients: RecipeIngredientModel[];
    steps: RecipeStepModel[];
} 

File: nom-api/Nom.Orch/Models/Recipe/RecipeEditModel.cs
// File: Nom.Orch/Models/Recipe/RecipeEditModel.cs

using System.Collections.Generic;

namespace Nom.Orch.Models.Recipe
{
    public class RecipeEditModel
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public List<RecipeIngredientModel> Ingredients { get; set; } = new List<RecipeIngredientModel>();
        public List<RecipeStepModel> Steps { get; set; } = new List<RecipeStepModel>();
    }
} 